CET: Agent onboarding wizard (#3800)
harness/developer-hub: docs/continuous-error-tracking/get-started/agent-onboarding-wizard.md

---
title: Agent Onboarding Wizard
description: Learn how to quickly intall and onboard an Error Tracking Agent.
sidebar_position: 70
---

> The CET Agent Onboarding Wizard helps you with quick installation and configuration of the CET Agent in your environment. It provides instructions for downloading the necessary Agent installation files. Additionally, it dynamically generates variable settings based on your environment details, enabling you to easily copy and apply them during the Agent installation process.

Reviewer: ```suggestion
The CET Agent Onboarding Wizard helps you quickly install and configure the CET Agent in your environment. The wizard:

* Provides instructions for downloading the necessary Agent installation files.
* Dynamically generates variable settings based on your environment details, which you can copy and apply during the Agent installation process.
```

Optional suggestion. Feel free to ignore.

This topic provides instructions on how to use the CET Agent Onboarding Wizard.

## Generate Agent configuration

To generate Agent configuration settings, do the following:

1. In your Harness project, navigate to the **Continuous Error Tracking** module, and then select **Monitored Services**.

> 2. The Monitored Services page appears, displaying a list of all the monitored services in your project. It also displays whether Agent and notifications are configured for each monitored service.

Reviewer: ```suggestion
   The Monitored Services page appears, displaying a list of all the monitored services in your project. It also displays whether Agent and notifications are configured for each monitored service.
```

This line doesn't contain instructions, just information about the page.

The step numbers need to be updated accordingly.

Author: Updated

> 3. Select **+ Configure Error Tracking Agent** to start creating a configuration. 

Reviewer: ```suggestion
2. Select **Configure Error Tracking Agent** to start creating a configuration. 
```

Not necessary to include the "nonverbal" part of the button unless it's required to distinguish it from another button with the same name.

   Alternatively, you can do the following:

    1. From the monitored services list, select the monitored service where you want to create a new agent configuration.

>     2. On the monitored service details page, go to the **Agent Configurations** tab, and then select **+ New error tracking agent configuration**.

Reviewer: At first glance, this looks like sub-steps. I would probably put this information in a tip note or a simple statement like "You can also select a monitored service, go to the **Agent Configurations** tab, and then select **New error tracking agent configuration** to launch the agent wizard."

Author: Updated

   
   The Configure error tracking agent for monitored service wizard appears.

>    <docimage path={require('./static/cet-agent-onboarding-wizard.png')} />

Reviewer: I think the image would make more sense if grouped with the step below since that is the first step where the user interacts with this screen.

4. Enter a name for the configuration and select **Next**.

5. Select a token from the dropdown and select **Next**.

> 6. Optionally provide source repository details if you wish to access code repository, branch, and commits related to errors from the ARC screen.

Reviewer: ```suggestion
6. Optionally, provide source repository details if you want to access a code repository, branch, and commits related to errors from the ARC screen.
```

> 7.  Select your application programming language, version, and OS. Based on the OS that you select, you may be asked to enter OS distribution details.  

Reviewer: ```suggestion
7.  Select your application programming language, version, and OS. Based on the OS that you select, you might need to enter OS distribution details.
```

   
   Once you have entered your programming language and OS information, CET displays the Agent configuration information and usage instructions.
 
8. Select **Save**.
   The saved configuration is listed in the **Agent Configurations** tab.

## Use an Agent configuration

> To view and use a configuration, follow these steps:

Reviewer: ```suggestion
To view and use a configuration, follow these steps:
```

It says "follow these steps" but there are no numbered steps after that.

Author: Updated

> - In the **Agent Configurations** tab, locate the configuration that you want to use and select the <i class="fa-solid fa-code"></i> button.  

Reviewer: Is there supposed to be a name here or is it literally a slash icon?

Author: It's literally a slash icon

>   The Configure error tracking agent for monitored service page is displayed. It provides instructions for downloading the Agent, adding JVM arguments, and setting environment variables.

Reviewer: ```suggestion
Use the instructions on the **Configure error tracking agent for monitored service** page to download the Agent, add JVM arguments, and set environment variables.
```

## Clone an Agent configuration

The **Clone** option allows you to create a new Agent configuration quickly based on an existing one.

To clone an Agent configuration, follow these steps:

> 1. In the **Agent Configurations** tab, locate the configuration you wish to clone and select the <i class="fa-regular fa-copy"></i> button.  

Reviewer: ```suggestion
1. In the **Agent Configurations** tab, locate the configuration you wish to clone and select the <i class="fa-regular fa-copy"></i> button.  
```

Since these are icons, it would help to have a text name along with them for accessibility reasons and in case the image doesn't render.

For example: 
`...select the **Copy** (<i class="fa-regular fa-copy"></i>) icon.`

Author: Updated

   
   The Configure error tracking agent for monitored service page is displayed.

2. Make the necessary changes to the configuration settings, and then select Save.

>    The newly saved configuration will appear in the Agent configuration tab.

Reviewer: ```suggestion
   The newly saved configuration appears in the Agent configuration tab.
```

## Edit an Agent configuration

The **Edit** option allows you to change the settings of an existing Agent configuration.

To edit an Agent configuration, follow these steps:

> 1. In the **Agent Configurations** tab, locate the configuration you wish to edit, and then select the ellipses (<i class="fa-solid fa-ellipsis-vertical"></i>) next to it. Then choose **Edit**.

Reviewer: ```suggestion
1. In the **Agent Configurations** tab, locate the configuration you wish to edit, and then select **More options** (<i class="fa-solid fa-ellipsis-vertical"></i>). Then choose **Edit**.
```

Brian and I have been calling this kebab menu "More options", since that is it's usual function in user interfaces in general.

   
   The Configure error tracking agent for monitored service page is displayed.

2. Make the necessary changes to the configuration settings, and then select **Save**.

   The changes will be applied to the Agent configuration.

## Delete an Agent configuration

The **Delete** option allows you to delete an existing Agent configuration.

To delete an Agent configuration, follow these steps:

> 1. In the **Agent Configurations** tab, locate the configuration you wish to edit, and then select the ellipses (<i class="fa-solid fa-ellipsis-vertical"></i>) next to it. Then choose **Delete**.

Reviewer: Same comment about More options instead of "ellipses".

Author: Updated
